src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
```

Comment:
Why do you need this **T** if you do not use it?

Reply:
I need add generic in my MachineProducer implementation:

https://imgur.com/a/7rShqKr

But I cannot do it, if I delete generic in MachineProducer interface. What do I make wrong?